packages/api/src/external/carequality/process-subsequent-outbound-patient-discovery-resps.ts

```diff
@@ -32,7 +32,8 @@ export async function processPostRespOutboundPatientDiscoveryResps({
 
     if (discoveryStatus !== "processing") {
       log(`Kicking off post resp patient discovery`);
-      await processPatientDiscoveryProgress({ patient, status: "processing" });
+      // BUG This will override any currently running status for PD (the bug carries over into the results endpoint as well)
+      await updatePatientDiscoveryStatus({ patient, status: "processing" });
```

Comment:
I made this a BUG not WARNING as I think this could be called fairly frequently.

Reply:
is this an old buggy behavior you are flagging or a new behavior as a result of the changes in this PR?

Reply:
Old buggy behavior. When it comes to updating the patient state, we've generally assumed in the code only one patient discovery workflow runs at a time for a given patient, but put in place no code to actually enforce this. I've started that process via the scheduling functionality, but technically this function needs a different fix as essentially it's "resurrecting" an already ended PD flow from the past, and we only store one PD flow status at a time right now, so we'd need somewhere else to tracking the status of this old PD flow is running again.

Reply:
Maybe I should create an issue for handling status tracking and add it to all the warnings / bugs I put in this PR.

Reply:
wait but doesnt the fact that updatePatientDiscoveryStatus wont run if discoveryStatus ==processing ensure that we dont override runs that are currently happening?

Reply:
You're right. By` currently running` I just meant another PD run generally (even if it's `completed` / `failed` -- not necessarily `processing`) -- sorry it's not clear in the comment. The fix here would be to reset the state of PD when this thing goes into `processing` so that "takes the lock" and does its thing. But, we've lost part of it's state by virtue of another PD kicking off (we do have it's request ID though). 

I removed the comment as flagging this issue everywhere doesn't seem worth it, and instead flagging it more generally seems like the right move.

Reply:
Larger state handling issue https://github.com/metriport/metriport-internal/issues/1832 (put a TODO in the code for each place right now we enter the PD flow outside of the scheduling framework).